doc/source/rllib/rllib-connector.rst

```diff
@@ -0,0 +1,272 @@
+            lambda actions, states, fetches: 2 * actions, states, fetches
+        )
+
+Mutiple connectors can be composed into a ``ConnectorPipeline``, which handles
+proper running of all children connectors in sequence and provides basic operations to modify and update the composition of connectors.
+
+``ConnectorPipeline`` also has agent and action versions:
+
+.. tabbed:: AgentConnectorPipeline
+
+    .. code-block:: python
+
+        # Example construction of an AgentConnectorPipeline.
+        pipeline = ActionConnectorPipeline(
+            ctx,
+            [ClipRewardAgentConnector(), ViewRequirementAgentConnector()]
+        )
+
+        # For demonstration purpose, we will add an ObsPreprocessorConnector
+        # in front of the ViewRequirementAgentConnector.
+        pipeline.insert_before("ViewRequirementAgentConnector", ObsPreprocessorConnector())
+
+.. tabbed:: Action Lambda Connector
+
+    .. code-block:: python
+
+        # Example construction of an ActionConnectorPipeline.
+        pipeline = ActionConnectorPipeline(
+            ctx,
+            [ConvertToNumpyConnector(), ClipActionsConnector(), ImmutableActionsConnector()]
+        )
+
+        # For demonstration purpose, we will drop the last ImmutableActionsConnector here.
+        pipeline.remove("ImmutableActionsConnector")
+
+
+
+Policy Checkpoint
+-----------------
+
+If connectors are enabled, RLlib will try to save policy checkpoints in properly serialized formats instead of
+relying on python pickling. Eventually, the goal is to save policy checkpoints in serialized JSON files
+to ensure maximum compatiiblity between RLlib and python versions.
+
+When enabled, the configurations of agent and action connectors will get serialized and saved with checkpointed
+policy states.
+These connectors, together with the specific transformations they represent,
+can be easily recovered (by RLlib provided utils) to simplify deployment and inference use cases.
+
+Serving and Inference
+---------------------
+
+With connectors essentially checkpointing all the transformations used during training,
+policies can be easily restored without the original algorithm for local inference,
+as demonstrated by the following Cartpole example:
+
+.. literalinclude:: ../../../rllib/examples/connectors/run_connector_policy.py
+   :language: python
+   :start-after: __sphinx_doc_begin__
+   :end-before: __sphinx_doc_end__
+
+RLlib will also provide utils soon to make server/client deployment of trained policies much easier.
+See `Notable TODOs`_.
+
+Adapting a Policy for Different Environments
+--------------------------------------------
+
+It not uncommon for user environments to go through active development iterations.
+Policies trained with an older version of an environment may be rendered useless for updated environments.
+While env wrapper helps with this problem in many cases, connectors allow policies trained with
+different environments to work together at a same time.
```

Comment:
...  together at the same time. would be correct I think.

Reply:
done